File: learning/katas/go/Core Transforms/GroupByKey/GroupByKey/test/task_test.go
// Licensed to the Apache Software Foundation (ASF) under one or more
// contributor license agreements.  See the NOTICE file distributed with
// this work for additional information regarding copyright ownership.
// The ASF licenses this file to You under the Apache License, Version 2.0
// (the "License"); you may not use this file except in compliance with
// the License.  You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

package test

import (
	"github.com/apache/beam/sdks/go/pkg/beam"
	"github.com/apache/beam/sdks/go/pkg/beam/testing/ptest"
	"groupbykey/pkg/task"
	"reflect"
	"testing"
)

func TestApplyTransform(t *testing.T) {
	p, s := beam.NewPipelineWithRoot()
	tests := []struct {
		input beam.PCollection
		want map[uint8][]string
	}{
		{
			input: beam.Create(s, "apple", "ball", "car", "bear", "cheetah", "ant"),
			want: map[uint8][]string{
				97: []string{"apple", "ant"},
				98: []string{"ball", "bear"},
				99: []string{"car", "cheetah"},
			},
		},
	}
	for _, tt := range tests {
		got := task.ApplyTransform(s, tt.input)
		beam.ParDo0(s, func(key uint8, values func(*string) bool) {
			var got []string
			var v string
			for values(&v) {
				got = append(got, v)
			}
			if !reflect.DeepEqual(got, tt.want[key]) {
				t.Errorf("ApplyTransform() = %v , want %v", got, tt.want[key])
			}
		}, got)
		if err := ptest.Run(p); err != nil {
			t.Error(err)
		}
	}
}

File: learning/katas/kotlin/Examples/Word Count/Word Count/test/org/apache/beam/learning/katas/examples/wordcount/TaskTest.kt
/*
 *  Licensed to the Apache Software Foundation (ASF) under one
 *  or more contributor license agreements.  See the NOTICE file
 *  distributed with this work for additional information
 *  regarding copyright ownership.  The ASF licenses this file
 *  to you under the Apache License, Version 2.0 (the
 *  "License"); you may not use this file except in compliance
 *  with the License.  You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */
package org.apache.beam.learning.katas.examples.wordcount

import org.apache.beam.learning.katas.examples.wordcount.Task.applyTransform
import org.apache.beam.sdk.testing.PAssert
import org.apache.beam.sdk.testing.TestPipeline
import org.apache.beam.sdk.transforms.Create
import org.junit.Rule
import org.junit.Test

class TaskTest {
  @get:Rule
  @Transient
  val testPipeline: TestPipeline = TestPipeline.create()

  @Test
  fun examples_word_count_word_count() {
    val lines = Create.of(
      "apple orange grape banana apple banana",
      "banana orange banana papaya"
    )

    val linesPColl = testPipeline.apply(lines)

    val results = applyTransform(linesPColl)

    PAssert.that(results).containsInAnyOrder(
      "apple:2",
      "banana:4",
      "grape:1",
      "orange:2",
      "papaya:1"
    )

    testPipeline.run().waitUntilFinish()
  }
}